medvis/dicom/src/io/gdcmvolumereader.cpp

```diff
@@ -0,0 +1,1070 @@
+        LogError("computed volume size is " << size << ", but stored is a volume of size " << len
+                                            << "!");
+    }
+
+    gdcm::MediaStorage ms = gdcm::MediaStorage();
+
+    // add some info in debug mode
+#if defined(IVW_DEBUG)
+    LogInfo("========================================================================");
+    LogInfo(file_ << " - Volume Information:");
+
+    std::string dimstr = std::to_string(dims[0]);
+    std::string oristr = std::to_string(origin[0]);
+    std::string spacestr = std::to_string(spacings[0]);
+    for (std::size_t i = 1; i < dims.size(); ++i) {
+        dimstr += "x" + std::to_string(dims[i]);
+        oristr += "x" + std::to_string(origin[i]);
+        spacestr += "x" + std::to_string(spacings[i]);
+    }
+
+    ms.SetFromFile(file);
+    gdcm::MediaStorage::MSType typ = ms.GetMSType(ms.GetString());
+
+    LogInfo("media storage type: " << ms.GetMSString(typ));
+    LogInfo("volume is " << dims.size() << "D");
+    LogInfo("size: " << dimstr);
+    LogInfo("offset: " << oristr);
+    LogInfo("spacing: " << spacestr);
+    LogInfo("volume size: " << size);
+    LogInfo("voxel size: " << voxelsz << "(components: " << (format->getComponents())
+                           << ", component size: " << (format->getSize()) << ")");
+    LogInfo("sample value range is [" << pixelformat.GetMin() << ", " << pixelformat.GetMax()
+                                      << "].");
+    pixelformat.Print(gdcm::Trace::GetDebugStream());
+    LogInfo("corresponding inviwo format: " << format->getString());
+    LogInfo("format: " << format->getString() << "[" << format->getMin() << ", " << format->getMax()
+                       << "]");
+    LogInfo("========================================================================");
+#endif
+
+    this->dimension_ = dimension;
+    this->format_ = format;
+    SharedVolume volume = std::make_shared<Volume>(dimension, format);
+    volume->setBasis(basis);
+    volume->setOffset(offset);
+    volume->setWorldMatrix(wtm);
+    volume->dataMap_.dataRange = dvec2(pixelformat.GetMin(), pixelformat.GetMax());
+
+    // image.GetSlope() and image.GetIntercept() can be incorrect (failure when calling
+    // image.Read()) Use dicom attributes director
+    std::vector<double> is = gdcm::ImageHelper::GetRescaleInterceptSlopeValue(file);
+    auto intercept = is[0];
+    auto slope = is[1];
+    const std::string modality(ms.GetModality());
+    auto maxValue = static_cast<double>(pixelformat.GetMax());
+
+    // TODO: check this heuristics!!!
+    if (modality == "CT") {
+        // Computed Tomography
+        volume->dataMap_.valueUnit = "HU";  // Hounsfield Unit
+        if (format->getPrecision() == 16) {
+            // Only show a subset (12-bit) of the data range
+            // Should be valid for scans of the human body
+            // where the range is [-1000 3095]
+            volume->dataMap_.dataRange = {0.0, 4095};
```

Comment:
Not so sure about this...

Reply:
Original code by @mskr, @VisComKreiser ...

Reply:
This function (GdcmVolumeReader::generateVolume) was actually written by someone else. We extended an existing DICOM reader, that was only able to read Mevis .dcm files. Maybe @VisComKreiser knows, where the reader came from.